src/group_elements.jl

```diff
@@ -1,223 +1,225 @@
     one(g::GroupElement)
-Return the identity element of the parent group of `g`.
+
+Return the identity element in the group of $g$.
 """
 Base.one(g::GroupElement) = one(parent(g))
 
 @doc Markdown.doc"""
-    order([BigInt, ]g::GroupElement)
-    order(I::Type{<:Integer}, g::GroupElement)
-Return the order of `g` as an instance of `I`.
+    order(::Type{I} = BigInt, g::GroupElement) where {I <: Integer}
 
-Only arbitrary size integers are required to return mathematically correct answer.
+Return the order of $g$ as an instance of $I$.
 """
-AbstractAlgebra.order(g::GroupElement) = order(BigInt, g)
+function order(::Type{I}, g::GroupElement) where {I<:Integer}
+    isfiniteorder(g) || throw(InfiniteOrder(g))
+    isone(g) && return I(1)
+    o = I(1)
+    gg = deepcopy(g)
+    out = similar(g)
+    while !isone(gg)
+        o += I(1)
+        gg = mul!(out, gg, g)
+    end
+    return o
+end
+order(g::GroupElement) = order(BigInt, g)
 
 @doc Markdown.doc"""
-    conj(g::GEl, h::GEl) where {GEl<:GroupElement}
-Return conjugation of `g` by `h`, i.e. `h^-1*g*h`.
+    conj(g::G, h::G) where {G <: GroupElement}
+
+Return conjugation of $g$ by $h$, i.e. $h^{-1} g h$.
 """
-Base.conj(g::GEl, h::GEl) where {GEl<:GroupElement} = conj!(similar(g), g, h)
+Base.conj(g::G, h::G) where {G <: GroupElement} = conj!(similar(g), g, h)
 
 @doc Markdown.doc"""
-    ^(g::GEl, h::GEl) where {GEl<:GroupElement}
-Conjugation action of `h` on `g`. See `conj`.
+    ^(g::G, h::G) where {G <: GroupElement}
+
+Alias for `conj`.
 """
-Base.:(^)(g::GEl, h::GEl) where {GEl<:GroupElement} = conj(g, h)
+Base.:(^)(g::G, h::G) where {G <: GroupElement} = conj(g, h)
 
 @doc Markdown.doc"""
-    comm(g::GEl, h::GEl[, Vararg{GEl}...) where GEl<:GroupElement
-Return the commutator `g^-1*h^-1*g*h` of `g` and `h`.
+    comm(g::G, h::G, k::G...) where {G <: GroupElement}
 
-The `Vararg` version returns the repeated (`foldl`) commutator, i.e.
-`comm(g, h, k) == comm(comm(g, h), k)`.
+Return the commutator $[[g, h], ...]$, where $[g, h] = g^{-1} h^{-1} g h$.
 """
-function comm(g::GEl, h::GEl, tail::GEl...) where {GEl<:GroupElement}
+function comm(g::G, h::G, k::G...) where {G <: GroupElement}
     res = comm!(similar(g), g, h)
-    for k in tail
+    for l in k
         res = comm!(res, res, k)
     end
     return res
 end
 
 Base.literal_pow(::typeof(^), g::GroupElement, ::Val{-1}) = inv(g)
 
-Base.:(/)(g::GEl, h::GEl) where {GEl<:GroupElement} =
-    div_right!(similar(g), g, h)
+@doc Markdown.doc"""
+    /(g::G, h::G) where {G <: GroupElement}
+
+Return $g h^{-1}$.
+"""
+Base.:(/)(g::G, h::G) where {G <: GroupElement} = div_right!(similar(g), g, h)
 
-#### Modification possible for performance reasons
+# NOTE: Modification possible for performance reasons
 
 @doc Markdown.doc"""
     similar(g::GroupElement)
-Return an arbitrary (and possibly uninitialized) group element sharing the parent with `g`.
+
+Return a group element sharing the parent with $g$. Might be arbitrary and
```

Comment:
I did not write that it does return an arbitrary element for obvious reasons (now the implementation is `one(g)`). Personally, I think it would be nice to emphasize that uninitialized elements are to be preferred for performance reasons.